x/dex/keeper/core_helper.go

```diff
@@ -521,3 +526,61 @@ func (k Keeper) BurnShares(ctx sdk.Context, addr sdk.AccAddress, amount sdk.Int,
 	}
 	return nil
 }
+
+///////////////////////////////////////////////////////////////////////////////
+//                            CORE VALIDATION                                //
+///////////////////////////////////////////////////////////////////////////////
+
+func (k Keeper) ValidateDeposit(
+	ctx sdk.Context,
+	callerAddr sdk.AccAddress,
+	feeTiers []types.FeeTier,
+	feeIndex uint64,
+	pair *types.TradingPair,
+	tickIndex int64,
+	amount0 sdk.Int,
+	amount1 sdk.Int,
+	balance0 sdk.Int,
+	balance1 sdk.Int,
```

Comment:
This signature is too big. Why are we separating this into a separate function?

Reply:
Seemed cleaner to me to have this in a separate function. Can move it back if you want.